components/MarkdownTextBlock/src/Renderers/ObjectRenderers/HtmlBlockRenderer.cs

```diff
@@ -0,0 +1,30 @@
+using HtmlAgilityPack;
+using Markdig.Syntax;
+
+namespace CommunityToolkit.Labs.WinUI.MarkdownTextBlock.Renderers.ObjectRenderers;
+
+internal class HtmlBlockRenderer : UWPObjectRenderer<HtmlBlock>
+{
+    protected override void Write(UWPRenderer renderer, HtmlBlock obj)
+    {
+        if (renderer == null) throw new ArgumentNullException(nameof(renderer));
+        if (obj == null) throw new ArgumentNullException(nameof(obj));
+
+        var stringBuilder = new StringBuilder();
+        foreach (var line in obj.Lines.Lines)
+        {
+            var lineText = line.Slice.ToString().Trim();
+            if (String.IsNullOrWhiteSpace(lineText))
+            {
+                continue;
+            }
+            stringBuilder.AppendLine(lineText);
+        }
+
+        var html = Regex.Replace(stringBuilder.ToString(), @"\t|\n|\r", "");
+        html = Regex.Replace(html, @"&nbsp;", " ");
```

Comment:
I guess you can precompile the regex? Can be done by using `RegexOptions.Compiled`.

Reply:
addressed